cpp/src/arrow/array/util.cc

```diff
@@ -915,6 +916,29 @@ Result<std::shared_ptr<Array>> MakeEmptyArray(std::shared_ptr<DataType> type,
   return builder->Finish();
 }
 
+Result<std::shared_ptr<Array>> MakeMaskArray(const std::vector<int64_t>& indices,
```

Comment:
This `std::vector` could be a `span<int64_t>` for more flexibility. The selection vector could come from anywhere.